### ISSUE TITLE
csi driver "secrets-store.csi.k8s.io" doesn't require volume_handle

### ISSUE THREAD

**Issue by giuseppeiannelli** [hashicorp/terraform-provider-kubernetes#1602]
### Terraform Version, Provider Version and Kubernetes Version

```
Terraform version:
Kubernetes provider version:
Kubernetes version:
```

### Terraform Configuration Files
```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.

terraform {
  # Do not specify maximum version on reusable modules as suggested here:
  # https://www.terraform.io/docs/language/providers/requirements.html#best-practices-for-provider-versions
  required_version = ">= 1.0"
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 3.0"
    }
    kubernetes = {
      source  = "hashicorp/kubernetes"
      version = "~> 2.0"
    }
    helm = {
      source  = "hashicorp/helm"
      version = "~> 2.0"
    }
  }
}

provider "aws" {}

data "aws_caller_identity" "current" {}

data "aws_eks_cluster" "this" {
  name = "changeme"
}

data "aws_eks_cluster_auth" "this" {
  name = data.aws_eks_cluster.this.name
}

provider "helm" {
  kubernetes {
    host                   = data.aws_eks_cluster.this.endpoint
    cluster_ca_certificate = base64decode(data.aws_eks_cluster.this.certificate_authority[0].data)
    token                  = data.aws_eks_cluster_auth.this.token
  }
}
provider "kubernetes" {
  host                   = data.aws_eks_cluster.this.endpoint
  cluster_ca_certificate = base64decode(data.aws_eks_cluster.this.certificate_authority[0].data)
  token                  = data.aws_eks_cluster_auth.this.token
}


locals {
  app_name          = "app-demo"
  cluster_id        = data.aws_eks_cluster.this.id
  oidc_provider     = replace(data.aws_eks_cluster.this.identity[0].oidc[0].issuer, "https://", "")
  oidc_provider_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/${local.oidc_provider}"
  kube_image = {
    repository = "busybox"
    tag        = "latest"
    pullPolicy = "Always"
  }
  kube_namespace = "kube-system"
  kube_labels = {
    "app.kubernetes.io/name"     = local.app_name
    "app.kubernetes.io/instance" = local.app_name
  }
  kube_labels_selector = merge(local.kube_labels, { "app.kubernetes.io/version" = local.kube_image.tag })
}

resource "aws_iam_role" "eks_iamserviceaccount_secrets_storage_csi" {
  name               = "eks-${local.cluster_id}-iamserviceaccount-secrets-storage-csi"
  assume_role_policy = data.aws_iam_policy_document.assume_role_iamserviceaccount_secrets_storage_csi.json
}

data "aws_iam_policy_document" "assume_role_iamserviceaccount_secrets_storage_csi" {
  statement {
    actions = ["sts:AssumeRoleWithWebIdentity"]
    effect  = "Allow"

    condition {
      test     = "StringEquals"
      variable = "${local.oidc_provider}:sub"
      values = [
        format(
          "system:serviceaccount:%s:%s",
          helm_release.secret_storage_csi_driver.metadata[0].namespace,
          "csi-secrets-store-provider-aws"
        )
      ]
    }

    principals {
      identifiers = [local.oidc_provider_arn]
      type        = "Federated"
    }
  }
}

# Install helm chart for CSI Secret Storage Driver
resource "helm_release" "secret_storage_csi_driver" {
  chart      = "secrets-store-csi-driver"
  repository = "https://kubernetes-sigs.github.io/secrets-store-csi-driver/charts"
  name       = "secrets-store-csi-driver"
  namespace  = local.kube_namespace
  version    = "1.0.1"
  atomic     = true

  set {
    name  = "linux.tolerations[0].operator"
    value = "Exists"
  }

  set {
    name  = "linux.tolerations[0].effect"
    value = "NoSchedule"
  }
}

#
# Stuff for AWS Secret Provisioner
#

resource "kubernetes_service_account" "csi_secrets_store_provider_aws" {
  metadata {
    name      = "csi-secrets-store-provider-aws"
    namespace = helm_release.secret_storage_csi_driver.metadata[0].namespace
    annotations = {
      "eks.amazonaws.com/role-arn" : aws_iam_role.eks_iamserviceaccount_secrets_storage_csi.arn
    }
  }
}

resource "kubernetes_cluster_role" "csi_secrets_store_provider_aws" {
  metadata {
    name = "csi-secrets-store-provider-aws-cluster-role"
  }
  rule {
    api_groups = [""]
    resources  = ["serviceaccounts/token"]
    verbs      = ["create"]
  }
  rule {
    api_groups = [""]
    resources  = ["serviceaccounts"]
    verbs      = ["get"]
  }
  rule {
    api_groups = [""]
    resources  = ["pods"]
    verbs      = ["get"]
  }
  rule {
    api_groups = [""]
    resources  = ["nodes"]
    verbs      = ["get"]
  }
}

resource "kubernetes_cluster_role_binding" "csi_secrets_store_provider_aws" {
  metadata {
    name = "csi-secrets-store-provider-aws-cluster-rolebinding"
  }
  role_ref {
    api_group = "rbac.authorization.k8s.io"
    kind      = "ClusterRole"
    name      = kubernetes_cluster_role.csi_secrets_store_provider_aws.metadata[0].name
  }
  subject {
    kind      = "ServiceAccount"
    name      = kubernetes_service_account.csi_secrets_store_provider_aws.metadata[0].name
    namespace = kubernetes_service_account.csi_secrets_store_provider_aws.metadata[0].namespace
  }
}

resource "kubernetes_daemonset" "csi_secrets_store_provider_aws" {
  metadata {
    name      = "csi-secrets-store-provider-aws"
    namespace = local.kube_namespace
    labels = {
      app = "csi-secrets-store-provider-aws"
    }
  }
  spec {
    strategy {
      rolling_update {}
    }
    selector {
      match_labels = {
        app = "csi-secrets-store-provider-aws"
      }
    }
    template {
      metadata {
        labels = {
          app = "csi-secrets-store-provider-aws"
        }
        annotations = {}
      }
      spec {
        service_account_name = kubernetes_service_account.csi_secrets_store_provider_aws.metadata[0].name
        host_network         = true
        container {
          name              = "provider-aws-installer"
          image             = "public.ecr.aws/aws-secrets-manager/secrets-store-csi-driver-provider-aws:1.0.r2-2021.08.13.20.34-linux-amd64"
          image_pull_policy = "Always"
          args = [
            "--provider-volume=/etc/kubernetes/secrets-store-csi-providers"
          ]
          resources {
            requests = {
              cpu    = "50m"
              memory = "100Mi"
            }
            limits = {
              cpu    = "50m"
              memory = "100Mi"
            }
          }
          volume_mount {
            mount_path = "etc/kubernetes/secrets-store-csi-providers"
            name       = "providervol"
          }
          volume_mount {
            mount_path        = "/var/lib/kubelet/pods"
            mount_propagation = "HostToContainer"
            name              = "mountpoint-dir"
          }
        }
        volume {
          name = "providervol"
          host_path {
            path = "/etc/kubernetes/secrets-store-csi-providers"
          }
        }
        volume {
          name = "mountpoint-dir"
          host_path {
            path = "/var/lib/kubelet/pods"
            type = "DirectoryOrCreate"
          }
        }
        node_selector = {
          "kubernetes.io/os" = "linux"
        }
        toleration {
          operator = "Exists"
          effect   = "NoSchedule"
        }
      }
    }
  }
}


resource "kubernetes_namespace_v1" "app" {
  metadata {
    name = "app"
  }
}

resource "aws_iam_role" "eks_iamserviceaccount_app" {
  name               = "eks-${local.cluster_id}-app-${local.app_name}-iamserviceaccount"
  assume_role_policy = data.aws_iam_policy_document.assume_role_iamserviceaccount.json
  managed_policy_arns = [
    aws_iam_policy.secret_storage_class.arn
  ]
}

data "aws_iam_policy_document" "assume_role_iamserviceaccount" {
  statement {
    actions = ["sts:AssumeRoleWithWebIdentity"]
    effect  = "Allow"

    condition {
      test     = "StringEquals"
      variable = "${local.oidc_provider}:sub"
      values = [
        format(
          "system:serviceaccount:%s:%s",
          kubernetes_namespace_v1.app.metadata[0].name,
          local.app_name
        )
      ]
    }

    principals {
      identifiers = [local.oidc_provider_arn]
      type        = "Federated"
    }
  }
}

data "aws_iam_policy_document" "secrets_storage_csi" {
  statement {
    effect = "Allow"
    actions = [
      "secretsmanager:GetSecretValue",
      "secretsmanager:DescribeSecret",
      "kms:Decrypt"
    ]
    resources = [
      aws_secretsmanager_secret.sensitive.arn,
      aws_kms_key.app.arn
    ]
  }
}

resource "aws_iam_policy" "secret_storage_class" {
  policy = data.aws_iam_policy_document.secrets_storage_csi.json
  name   = "app-${local.app_name}-access-to-sensitive"
}

resource "kubernetes_service_account_v1" "this" {
  metadata {
    name      = local.app_name
    namespace = kubernetes_namespace_v1.app.metadata[0].name
    labels    = local.kube_labels
    annotations = {
      "eks.amazonaws.com/role-arn" = aws_iam_role.eks_iamserviceaccount_app.arn
    }
  }
}

resource "aws_kms_key" "app" {
  deletion_window_in_days = 7
  description             = "app sensitive data encryption key for ${local.app_name}"
}

resource "aws_kms_alias" "app" {
  target_key_id = aws_kms_key.app.id
  name          = "alias/app/${local.app_name}"
}

resource "aws_secretsmanager_secret" "sensitive" {
  name                    = "/app/${local.app_name}/sensitive"
  kms_key_id              = aws_kms_key.app.id
  recovery_window_in_days = 7
}

resource "aws_secretsmanager_secret_version" "sensitive" {
  secret_id     = aws_secretsmanager_secret.sensitive.id
  secret_string = "secretstringdata"
}

resource "kubernetes_manifest" "secret-provider-class" {
  manifest = {
    apiVersion = "secrets-store.csi.x-k8s.io/v1"
    kind       = "SecretProviderClass"
    metadata = {
      namespace = kubernetes_namespace_v1.app.metadata[0].name
      name      = local.app_name
    }
    spec = {
      provider = "aws"
      parameters = {
        objects = yamlencode([
          {
            objectName  = aws_secretsmanager_secret.sensitive.id
            objectType  = "secretsmanager"
            objectAlias = "sensitive"
          }
        ])
      }
    }
  }
}

resource "kubernetes_pod_v1" "app" {
  metadata {
    name      = local.app_name
    namespace = kubernetes_namespace_v1.app.metadata[0].name
    labels    = local.kube_labels

  }
  spec {
    container {
      name = local.app_name
      security_context {
        capabilities {
          drop = [
            "ALL"
          ]
        }
        privileged                = false
        read_only_root_filesystem = true
        run_as_group              = "65534"
        run_as_non_root           = true
        run_as_user               = "65534"
      }
      image             = "${local.kube_image.repository}:${local.kube_image.tag}"
      image_pull_policy = local.kube_image.pullPolicy
      volume_mount {
        mount_path = "/var/run/secrets/app/sensitive"
        name       = "sensitive"
      }
    }
    security_context {
      fs_group        = "65534"
      run_as_group    = "65534"
      run_as_non_root = true
      run_as_user     = "65534"
    }
    service_account_name = kubernetes_service_account_v1.this.metadata[0].name
    volume {
      name = "sensitive"
      csi {
        driver    = "secrets-store.csi.k8s.io"
        read_only = true
        #        volume_handle = ""
        volume_attributes = {
          secretProviderClass = kubernetes_manifest.secret-provider-class.manifest.metadata.name
        }
      }
    }
  }
}
```

### Steps to Reproduce

1. Change EKS cluster name
2. `terraform apply` 

### Expected Behavior
What should have happened?

I want to create a deployment that use a CSI volume from `secrets-store.csi.k8s.io` driver.
The `secrets-store.csi.k8s.io` driver doesn't require `volume_handle`

### Actual Behavior
What actually happened?

`volume_handle` param is required for `csi` volume type

### References

- https://github.com/aws/secrets-store-csi-driver-provider-aws/tree/r2/examples
- https://secrets-store-csi-driver.sigs.k8s.io/
- https://github.com/kubernetes-sigs/secrets-store-csi-driver

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by pdrd** [comment]
Experiencing the same issue with Terraform v1.1.5 and provider hashicorp/kubernetes v2.8.0 with kubernetes v1.21.7.

Setting any value for `volume_handle` like empty string or a non-existing value like `"kv"` results in a replacement of the `csi {}` block with a `empty_dir {}` block in the volume definition. Therefore we are currently not able to create a kubernetes deployment with a csi secrets store.

@giuseppeiannelli Are you aware of any workaround?

**Comment by alexsomesan** [comment]
Folks, please share TF configuration that we can use to try to reproduce the issue. 
It really helps us get to the solution faster when we don't have to invent and assume what the user might have been doing when they ran into the problem.

**Comment by dghubble** [comment]
This is the same issue as https://github.com/hashicorp/terraform-provider-kubernetes/pull/1092

**Comment by SebastianOpriel** [comment]
I am looking for a solution to this problem too. So: +1

**Comment by alinefr** [comment]
At the moment I'm running my own private fork k8s provider based on #1092, but it's very hard to keep it in sync with upstream.
I need CSI to keep k8s secrets in AWS Secrets Manager. It would be awesome if you guys could have a look in the proposed PR in #1092 or provide a better fix. Thanks!

**Comment by giuseppeiannelli** [comment]
@pdrd I don't any workaround for this issue :(

**Comment by giuseppeiannelli** [comment]
@alexsomesan Following the configuration to install the CSI driver on some EKS cluster and reproduce the issue.

```hcl
terraform {
  # Do not specify maximum version on reusable modules as suggested here:
  # https://www.terraform.io/docs/language/providers/requirements.html#best-practices-for-provider-versions
  required_version = ">= 1.0"
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 3.0"
    }
    kubernetes = {
      source  = "hashicorp/kubernetes"
      version = "~> 2.0"
    }
    helm = {
      source  = "hashicorp/helm"
      version = "~> 2.0"
    }
  }
}

provider "aws" {}

data "aws_caller_identity" "current" {}

data "aws_eks_cluster" "this" {
  name = "changeme"
}

data "aws_eks_cluster_auth" "this" {
  name = data.aws_eks_cluster.this.name
}

provider "helm" {
  kubernetes {
    host                   = data.aws_eks_cluster.this.endpoint
    cluster_ca_certificate = base64decode(data.aws_eks_cluster.this.certificate_authority[0].data)
    token                  = data.aws_eks_cluster_auth.this.token
  }
}
provider "kubernetes" {
  host                   = data.aws_eks_cluster.this.endpoint
  cluster_ca_certificate = base64decode(data.aws_eks_cluster.this.certificate_authority[0].data)
  token                  = data.aws_eks_cluster_auth.this.token
}


locals {
  app_name          = "app-demo"
  cluster_id        = data.aws_eks_cluster.this.id
  oidc_provider     = replace(data.aws_eks_cluster.this.identity[0].oidc[0].issuer, "https://", "")
  oidc_provider_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/${local.oidc_provider}"
  kube_image = {
    repository = "busybox"
    tag        = "latest"
    pullPolicy = "Always"
  }
  kube_namespace = "kube-system"
  kube_labels = {
    "app.kubernetes.io/name"     = local.app_name
    "app.kubernetes.io/instance" = local.app_name
  }
  kube_labels_selector = merge(local.kube_labels, { "app.kubernetes.io/version" = local.kube_image.tag })
}

resource "aws_iam_role" "eks_iamserviceaccount_secrets_storage_csi" {
  name               = "eks-${local.cluster_id}-iamserviceaccount-secrets-storage-csi"
  assume_role_policy = data.aws_iam_policy_document.assume_role_iamserviceaccount_secrets_storage_csi.json
}

data "aws_iam_policy_document" "assume_role_iamserviceaccount_secrets_storage_csi" {
  statement {
    actions = ["sts:AssumeRoleWithWebIdentity"]
    effect  = "Allow"

    condition {
      test     = "StringEquals"
      variable = "${local.oidc_provider}:sub"
      values = [
        format(
          "system:serviceaccount:%s:%s",
          helm_release.secret_storage_csi_driver.metadata[0].namespace,
          "csi-secrets-store-provider-aws"
        )
      ]
    }

    principals {
      identifiers = [local.oidc_provider_arn]
      type        = "Federated"
    }
  }
}

# Install helm chart for CSI Secret Storage Driver
resource "helm_release" "secret_storage_csi_driver" {
  chart      = "secrets-store-csi-driver"
  repository = "https://kubernetes-sigs.github.io/secrets-store-csi-driver/charts"
  name       = "secrets-store-csi-driver"
  namespace  = local.kube_namespace
  version    = "1.0.1"
  atomic     = true

  set {
    name  = "linux.tolerations[0].operator"
    value = "Exists"
  }

  set {
    name  = "linux.tolerations[0].effect"
    value = "NoSchedule"
  }



}

#
# Stuff for AWS Secret Provisioner
#

resource "kubernetes_service_account" "csi_secrets_store_provider_aws" {
  metadata {
    name      = "csi-secrets-store-provider-aws"
    namespace = helm_release.secret_storage_csi_driver.metadata[0].namespace
    annotations = {
      "eks.amazonaws.com/role-arn" : aws_iam_role.eks_iamserviceaccount_secrets_storage_csi.arn
    }
  }


}

resource "kubernetes_cluster_role" "csi_secrets_store_provider_aws" {
  metadata {
    name = "csi-secrets-store-provider-aws-cluster-role"
  }
  rule {
    api_groups = [""]
    resources  = ["serviceaccounts/token"]
    verbs      = ["create"]
  }
  rule {
    api_groups = [""]
    resources  = ["serviceaccounts"]
    verbs      = ["get"]
  }
  rule {
    api_groups = [""]
    resources  = ["pods"]
    verbs      = ["get"]
  }
  rule {
    api_groups = [""]
    resources  = ["nodes"]
    verbs      = ["get"]
  }


}

resource "kubernetes_cluster_role_binding" "csi_secrets_store_provider_aws" {
  metadata {
    name = "csi-secrets-store-provider-aws-cluster-rolebinding"
  }
  role_ref {
    api_group = "rbac.authorization.k8s.io"
    kind      = "ClusterRole"
    name      = kubernetes_cluster_role.csi_secrets_store_provider_aws.metadata[0].name
  }
  subject {
    kind      = "ServiceAccount"
    name      = kubernetes_service_account.csi_secrets_store_provider_aws.metadata[0].name
    namespace = kubernetes_service_account.csi_secrets_store_provider_aws.metadata[0].namespace
  }


}

resource "kubernetes_daemonset" "csi_secrets_store_provider_aws" {
  metadata {
    name      = "csi-secrets-store-provider-aws"
    namespace = local.kube_namespace
    labels = {
      app = "csi-secrets-store-provider-aws"
    }
  }
  spec {
    strategy {
      rolling_update {}
    }
    selector {
      match_labels = {
        app = "csi-secrets-store-provider-aws"
      }
    }
    template {
      metadata {
        labels = {
          app = "csi-secrets-store-provider-aws"
        }
        annotations = {}
      }
      spec {
        service_account_name = kubernetes_service_account.csi_secrets_store_provider_aws.metadata[0].name
        host_network         = true
        container {
          name              = "provider-aws-installer"
          image             = "public.ecr.aws/aws-secrets-manager/secrets-store-csi-driver-provider-aws:1.0.r2-2021.08.13.20.34-linux-amd64"
          image_pull_policy = "Always"
          args = [
            "--provider-volume=/etc/kubernetes/secrets-store-csi-providers"
          ]
          resources {
            requests = {
              cpu    = "50m"
              memory = "100Mi"
            }
            limits = {
              cpu    = "50m"
              memory = "100Mi"
            }
          }
          volume_mount {
            mount_path = "etc/kubernetes/secrets-store-csi-providers"
            name       = "providervol"
          }
          volume_mount {
            mount_path        = "/var/lib/kubelet/pods"
            mount_propagation = "HostToContainer"
            name              = "mountpoint-dir"
          }
        }
        volume {
          name = "providervol"
          host_path {
            path = "/etc/kubernetes/secrets-store-csi-providers"
          }
        }
        volume {
          name = "mountpoint-dir"
          host_path {
            path = "/var/lib/kubelet/pods"
            type = "DirectoryOrCreate"
          }
        }
        node_selector = {
          "kubernetes.io/os" = "linux"
        }
        toleration {
          operator = "Exists"
          effect   = "NoSchedule"
        }
      }
    }
  }


}


resource "kubernetes_namespace_v1" "app" {
  metadata {
    name = "app"
  }


}

resource "aws_iam_role" "eks_iamserviceaccount_app" {
  name               = "eks-${local.cluster_id}-app-${local.app_name}-iamserviceaccount"
  assume_role_policy = data.aws_iam_policy_document.assume_role_iamserviceaccount.json
  managed_policy_arns = [
    aws_iam_policy.secret_storage_class.arn
  ]
}

data "aws_iam_policy_document" "assume_role_iamserviceaccount" {
  statement {
    actions = ["sts:AssumeRoleWithWebIdentity"]
    effect  = "Allow"

    condition {
      test     = "StringEquals"
      variable = "${local.oidc_provider}:sub"
      values = [
        format(
          "system:serviceaccount:%s:%s",
          kubernetes_namespace_v1.app.metadata[0].name,
          local.app_name
        )
      ]
    }

    principals {
      identifiers = [local.oidc_provider_arn]
      type        = "Federated"
    }
  }


}

data "aws_iam_policy_document" "secrets_storage_csi" {
  statement {
    effect = "Allow"
    actions = [
      "secretsmanager:GetSecretValue",
      "secretsmanager:DescribeSecret",
      "kms:Decrypt"
    ]
    resources = [
      aws_secretsmanager_secret.sensitive.arn,
      aws_kms_key.app.arn
    ]
  }
}

resource "aws_iam_policy" "secret_storage_class" {
  policy = data.aws_iam_policy_document.secrets_storage_csi.json
  name   = "app-${local.app_name}-access-to-sensitive"
}

resource "kubernetes_service_account_v1" "this" {
  metadata {
    name      = local.app_name
    namespace = kubernetes_namespace_v1.app.metadata[0].name
    labels    = local.kube_labels
    annotations = {
      "eks.amazonaws.com/role-arn" = aws_iam_role.eks_iamserviceaccount_app.arn
    }
  }


}

resource "aws_kms_key" "app" {
  deletion_window_in_days = 7
  description             = "app sensitive data encryption key for ${local.app_name}"
}

resource "aws_kms_alias" "app" {
  target_key_id = aws_kms_key.app.id
  name          = "alias/app/${local.app_name}"
}

resource "aws_secretsmanager_secret" "sensitive" {
  name                    = "/app/${local.app_name}/sensitive"
  kms_key_id              = aws_kms_key.app.id
  recovery_window_in_days = 7
}

resource "aws_secretsmanager_secret_version" "sensitive" {
  secret_id     = aws_secretsmanager_secret.sensitive.id
  secret_string = "secretstringdata"
}

resource "kubernetes_manifest" "secret-provider-class" {
  manifest = {
    apiVersion = "secrets-store.csi.x-k8s.io/v1"
    kind       = "SecretProviderClass"
    metadata = {
      namespace = kubernetes_namespace_v1.app.metadata[0].name
      name      = local.app_name
    }
    spec = {
      provider = "aws"
      parameters = {
        objects = yamlencode([
          {
            objectName  = aws_secretsmanager_secret.sensitive.id
            objectType  = "secretsmanager"
            objectAlias = "sensitive"
          }
        ])
      }
    }
  }



}

resource "kubernetes_pod_v1" "app" {
  metadata {
    name      = local.app_name
    namespace = kubernetes_namespace_v1.app.metadata[0].name
    labels    = local.kube_labels

  }
  spec {
    container {
      name = local.app_name
      security_context {
        capabilities {
          drop = [
            "ALL"
          ]
        }
        privileged                = false
        read_only_root_filesystem = true
        run_as_group              = "65534"
        run_as_non_root           = true
        run_as_user               = "65534"
      }
      image             = "${local.kube_image.repository}:${local.kube_image.tag}"
      image_pull_policy = local.kube_image.pullPolicy
      volume_mount {
        mount_path = "/var/run/secrets/app/sensitive"
        name       = "sensitive"
      }
    }
    security_context {
      fs_group        = "65534"
      run_as_group    = "65534"
      run_as_non_root = true
      run_as_user     = "65534"
    }
    service_account_name = kubernetes_service_account_v1.this.metadata[0].name
    volume {
      name = "sensitive"
      csi {
        driver    = "secrets-store.csi.k8s.io"
        read_only = true
        #        volume_handle = ""
        volume_attributes = {
          secretProviderClass = kubernetes_manifest.secret-provider-class.manifest.metadata.name
        }
      }
    }
  }
}
```

**Comment by arybolovlev** [comment]
Hi @giuseppeiannelli ,

The fix for this issue #1092 has been merged into the main branch and will be available in the upcoming release. Thank you for reporting this issue and the detailed description.

I will go ahead and close this issue. Please keep an eye on releases and release notes.

Thanks.
